docs/electrical/embedded/styleguide.md

```diff
@@ -1,3 +1,190 @@
+2. Unless specified by another whitespace rule, all operators should be surrounded on either side by a single space.
+3. Unary operators should not have a space between the operator and the operand.
+4. Structure pointer and member operators should have no whitespace on either side.
+5. Parentheses, braces, and brackets should not have an adjacent space on the interior side.
+6. Excluding the end of a line, separating commas should should be followed by one space.
+7. Semicolons should follow a preceeding statement without a space, but should always be followed by a space unless they end a line.
+8. Use whitespace to separate blocks of code from each other in a logical fashion. (William's note: to please my OCD use either 1 or 3 lines of separation depending on context, don't use two :) )
+
+#### Pointer (* and &) Operators
+
+1. Pointer operators should be written without a space on the operand side (i.e. ```type *name```).
+2. Pointer variables should be declared on a line separate from non-pointer variables of the same type, to avoid confusion between the two.
+
+#### Alignment Rules
+
+1. TODO: Consider whether to always left align the # of a preprocessor directive, regardless of the indentation of the line
+2. Align relevant blocks of code together (Don't be stupid) 
+    TODO: Make this nicer?
+
+#### Tab Rules
+
+1. The unicode tab character should not appear in files. Instead, the specified tab size is 4 spaces.
+
+### Module Rules
+
+#### Naming Conventions
+
+1. File names should consist only of lower case letters, numbers, and underscores.
+2. File names are recommended to be no longer than 16 characters. Ideally file names are specific and short, like the file itself.
+3. All module names should not overlap with those of the standard library.
+4. All files that contain a ```main()``` function should include "main" within the filename.
+
+#### Header Files
+
+1. Include preprocessor guards against multiple inclusion within all header files (#ifndef NAME_H)
+2. No variables should be declared within a header file, nor should any storage be allocated
+3. Header files should concern themselves strictly with the information that is needed to be known by other modules. (Everything else should be declared within the source file itself)
+
+#### Source Files
+
+1. Source files should be specific, and contain only the information relevant to controlling one 'domain' (UART driver, Servo control, etc).
+2. Source files should always ```#include``` the corresponding header file.
+3. No absolute paths should be used in ```#include``` statements
+4. Do not include extraneous ```#include``` statements.
+5. Do not directly ```#include``` another source file; always ```#include``` the relevant header file.
+6. Any variables shared internally across the entire module but that aren't declared according to the rules of the header file should be declared at the top of the module's source file.
+
+### Data Type Rules
+
+#### Type Definitions
```

Comment:
Consider a rule for structs. Should structs be typedef'd or no?

With typedef (my preference; I think it makes the code more readable)

```c
typedef struct {
    int a;
    int b;
} mystruct_t;

int main(void){
    mystruct_t myvar;
}
```

Without typedef

```c
struct mystruct {
    int a;
    int b;
};

int main(void){
    struct mystruct myvar;
}
```

The argument for the latter is that it makes it "more clear" that myvar is a struct. However, I dislike the syntax as it seems to be a confusing things for less experienced developers.